src/Stratis.Features.SQLiteWalletRepository/Tables/HDTransactionData.cs

```diff
@@ -231,105 +232,103 @@ FROM    HDTransactionData
+                    GROUP   BY t.OutputTxId
+                    UNION   ALL";
 
-            -- Interwoven receives and spends
-SELECT * FROM
-            (
-              -- Find all receives
-              SELECT
-                  t.OutputTxId as Id,
-                  t.RedeemScript,
-                CASE 
-                    WHEN t.OutputTxIsCoinbase = 0 AND t.AddressType = 0 THEN 0
-                    WHEN t.OutputTxIsCoinbase = 1 AND t.OutputIndex = 0 THEN 3
-                    WHEN t.OutputTxIsCoinbase = 1 AND t.OutputIndex != 0 THEN 2
-                END Type,                    
-                t.OutputTxTime as TimeStamp,
-                CASE
-                    WHEN t.OutputTxIsCoinbase = 0 AND t.AddressType = 0 THEN t.Value
-                    WHEN t.OutputTxIsCoinbase = 0 AND t.AddressType = 1 THEN ((SELECT sum(tt.Value) FROM HDTransactionData tt WHERE tt.SpendTxId = t.OutputTxId) - t.Value)
-                    WHEN t.OutputTxIsCoinbase = 1 AND t.OutputIndex = 0 THEN t.Value
-                    WHEN t.OutputTxIsCoinbase = 1 AND t.OutputIndex != 0 THEN (SUM(t.Value) -
-                    (
-                        SELECT ttp.Value
-                        FROM HDPayment p
-                        INNER JOIN HDTransactionData ttp ON ttp.OutputTxId = p.OutputTxId AND ttp.OutputIndex = p.OutputIndex AND ttp.WalletId = {strWalletId} AND ttp.AccountIndex = {strAccountIndex}
-                        WHERE p.SpendTxId = t.OutputTxId AND p.SpendIsChange = 0
-                        LIMIT 1
-                    ))
-                END Amount,
-                NULL as Fee,
-                NULL as SendToScriptPubkey,
-                t.Address AS ReceiveAddress,
-                t.OutputBlockHeight as BlockHeight
-              FROM 
-                HDTransactionData AS t
-              WHERE t.WalletId = {strWalletId} AND t.AccountIndex = {strAccountIndex}
-              GROUP BY t.OutputTxId
-            UNION ALL
-                SELECT * FROM 
-                (
-                    SELECT
-                    		t.SpendTxId as Id,
-                    		t.RedeemScript,
-                    		1 as Type,
-                    		t.SpendTxTime as TimeStamp,
-                    		IFNULL(p.SendValue, 0) AS Amount,
-                    		t.Fee,
-                    		p.SpendScriptPubKey as Address,
-                    		NULL AS ReceiveAddress,
-                    		t.SpendBlockHeight as BlockHeight
+            string spends = $@"
+                    -- Find all spends
+                    SELECT  t.SpendTxId as Id,
+                    	    t.RedeemScript,
+                    	    1 as Type,
+                    	    t.SpendTxTime as TimeStamp,
+                    	    IFNULL(p.SendValue, 0) AS Amount,
+                    	    t.Fee,
+                    	    p.SpendScriptPubKey as Address,
```

Comment:
Just double checking this... The controller method receives a address and not the ScriptPubKey representation of it. Should we "convert" the address we filter against to the script pub key here (or earlier) ?

Reply:
That is the `SentToScriptPubKey` (according to the first query of the UNION ALL) and its converted here:
![image](https://user-images.githubusercontent.com/29645989/123918779-8e0f0380-d9c7-11eb-9339-eb1fe53c74e8.png)

I am assuming that address filtering is for isolating a specific address in an account so I think in that case we would be using the `HD_TransactionData`'s `Address` field to compare against?